### PR TITLE
be/fixes: add users

### PR DESCRIPTION
**Pull request by Nasfame**
### Description

- [x] Resolves  

### Checklist

- [x] I have followed the repository's contribution guidelines.
- [x] I have added necessary documentation or updated existing documentation.
- [x] My code follows the established code style and formatting of the project.
- [x] I have tested my changes and they function as expected.
- [x] I have added suitable unit tests (if applicable).

To be reviewed by: @Nasfame

### CODE REVIEW

**Review by techyNonso — COMMENTED** [review]
image

Victor said users can also register with meta mask and in such case they will not have email at the begining.



I this suggest we just work with google and email for now then later we integrate metamask cos this will lead to re writing the schema and changing frontend code base to support that.

**Review by techyNonso — CHANGES_REQUESTED** [review]
image

This should returned created or updated user, we then set context with with the created user cos we need the user ID

**Comment by Nasfame** [comment]
> image
> 
> Victor said users can also register with meta mask and in such case they will not have email at the begining.
> 
> I this suggest we just work with google and email for now then later we integrate metamask cos this will lead to re writing the schema and changing frontend code base to support that.

u realize the mongo - is no sql (no schema) , we can always change the schema

**Comment by techyNonso** [comment]
> > image
> > Victor said users can also register with meta mask and in such case they will not have email at the begining.
> > I this suggest we just work with google and email for now then later we integrate metamask cos this will lead to re writing the schema and changing frontend code base to support that.
> 
> u realize the mongo - is no sql (no schema) , we can always change the schema

Yea I know but for those without email, we will have to use wallet address for checking if user exists .

**Comment by techyNonso** [comment]
Referring to the upsert call

File: src/app/explore/page.tsx [hunk 1]
```diff
@@ -1,106 +1,102 @@
...
+        setIsLoading(true);
+        push('/dashboard');
       }
-    }
-    checkStatus()
-    return
-  }, [])
+    };
+    checkStatus();
+  }, []);
```
[the remark this answers is not in the document]
Comment (techyNonso) [reply src/app/explore/page.tsx:74]: Thought we removed this

[the remark this answers is not in the document]
Comment (Nasfame) [reply src/app/explore/page.tsx:74]: no we didn't remove this @techyNonso

**Review by techyNonso — APPROVED** [review]
LGTM!

**Pull request merged**